service/worker/migration/activities.go

```diff
@@ -482,3 +489,120 @@ func (a *activities) SeedReplicationQueueWithUserDataEntries(ctx context.Context
 		activity.RecordHeartbeat(ctx, heartbeatDetails)
 	}
 }
+
+func (a *activities) verifyReplicationTasks(ctx context.Context, request *genearteAndVerifyReplicationTasksRequest, verifieldFlags []bool, adminClient adminservice.AdminServiceClient) (bool, error) {
+	var verified = true
+	for i := 0; i < len(request.Executions); i++ {
+		if verifieldFlags[i] {
+			continue
+		}
+
+		we := request.Executions[i]
+		if _, err := adminClient.DescribeMutableState(ctx, &adminservice.DescribeMutableStateRequest{
+			Namespace: request.Namespace,
+			Execution: &we,
+		}); err != nil {
+			var notFoundErr *serviceerror.NotFound
+			if !errors.As(err, &notFoundErr) {
+				return false, err
+			}
+
+			// Continue to verify even though a single workflow execution was not found.
+			verified = false
+		} else {
+			a.metricsHandler.Counter(metrics.ForceReplicationVerifyReplicationSuccess.GetMetricName()).Record(1)
+			verifieldFlags[i] = true
+		}
+	}
+
+	return verified, nil
+}
+
+type (
+	replicationTasksHeartbeatDetails struct {
+		VerifiedFlags      []bool
+		VerifyTimeoutCount int
+	}
+
+	verifyReplicationTasksTimeoutErr struct {
+		Details replicationTasksHeartbeatDetails
+	}
+)
+
+func (e verifyReplicationTasksTimeoutErr) Error() string {
+	return fmt.Sprintf("Failed to verify replication tasks. Details: %v", e.Details)
+}
+
+func (a *activities) GenerateAndVerifyReplicationTasks(ctx context.Context, request *genearteAndVerifyReplicationTasksRequest) error {
+	ctx = a.setCallerInfoForGenReplicationTask(ctx, namespace.ID(request.NamespaceID))
+	rateLimiter := quotas.NewRateLimiter(request.RPS, int(math.Ceil(request.RPS)))
+
+	var details replicationTasksHeartbeatDetails
+	if activity.HasHeartbeatDetails(ctx) {
+		if err := activity.GetHeartbeatDetails(ctx, &details); err != nil {
+			return err
+		}
+	}
+
+	if len(details.VerifiedFlags) == 0 {
+		details.VerifiedFlags = make([]bool, len(request.Executions))
+		activity.RecordHeartbeat(ctx, details)
+	}
+
+	var executions []commonpb.WorkflowExecution
+	for i := 0; i < len(request.Executions); i++ {
+		if !details.VerifiedFlags[i] {
+			executions = append(executions, request.Executions[i])
+		}
+	}
+
+	generateErrChan := make(chan error)
+	go func() {
+		for _, we := range executions {
+			err := a.generateWorkflowReplicationTask(ctx, rateLimiter, definition.NewWorkflowKey(request.NamespaceID, we.WorkflowId, we.RunId))
```

Comment:
we ignore the not found error and I think if it returns not found, we don't need to verify it.